kubespawner/spawner.py

```diff
@@ -2662,6 +2660,8 @@ async def _make_create_resource_request(self, kind, manifest):
 
     async def _start(self):
         """Start the user's pod"""
+        pvc_name = self._expand_user_properties(self.pvc_name_template)
```

Comment:
I've been looking at this in #744, and `pvc_name` should actually be one of the things persisted in Spawner state, because the pvc name should probably be preserved to avoid data loss even if it. Resolving the template here would ensure the new template is used and the old pvc is orphaned. But that's _intended_ when overrides are used as in #761.

Should we:

1. not try to track existing pvcs (status quo, this PR), or
2. try not to lose data (what I'm trying to do in #744)

If we should try not to lose data, how do we distinguish between a changed pvc for a profile (should change) vs config (should _not_ change if prior pvc already created)?